### ISSUE TITLE
This should be part of whatwg/html

### ISSUE THREAD

**Issue by annevk** [InternetExplorer/directory-upload#15]
Defining this standalone seems to miss all kinds of subtleties that would be more self-evident if this were a PR to whatwg/html. In particular the interaction with all the other input states is rather hand wavy, but there's also cause for concern with future updates to `<input type=file>.files`.

**Comment by travisleithead** [comment]
Let's not rule that out as an option, but starting this in the incubator group is the right way to seed this proposal. Do you have specific concerns around the interation with other input states (and state transitions)?

**Comment by annevk** [comment]
I disagree with "the right way", especially considering how the proposal doesn't integrate well with how form features have been defined to date. I filed a number of more specific issues, though I only reviewed for a couple of minutes total.

**Comment by domenic** [comment]
In general the spec as-is does not separate out pieces clearly. I've mentioned some of the issues over in #17, but it patches MANY different places of the spec _in the definition of the directory IDL attribute_, which is crazy. It patches [rendering](https://html.spec.whatwg.org/#the-input-element-as-a-file-upload-control). It patches the `files` IDL attribute. It patches the activation behavior, without concern for the mutable vs. immutable state. And it talks about concepts like " triggers the default click behavior of the input element," which do not exist.

This is again just a five-minute review.

An actual patch would turn up a lot of these issues, as it tried to fit into the existing framework. The PR review by the editors would help fix the rest.

**Comment by inexorabletash** [comment]
ISTM this would need to be patches for both HTML and FILEAPI (to add the Directory interface) - HTML already has a dependency on FILEAPI for File/FileList.

IMHO, that would help sort out #19 as well - drop a dependency on FileSystem and avoid adding a third spec to the mix prematurely.

**Comment by arunranga** [comment]
(EDITED) @inexorabletash : Rather than add / patch File API, we could have a separate proposal (Directory Upload) that frees us from a dependency on FileSystem API. 

FileSystem API, when implementers get to it, can actually build on Directory Upload. That would better match the order of implementation.
